anomalib/models/components/feature_extractors/__init__.py

```diff
@@ -3,14 +3,142 @@
 # Copyright (C) 2022 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-from .timm import FeatureExtractor, TimmFeatureExtractor
-from .torchfx import BackboneParams, TorchFXFeatureExtractor
-from .utils import dryrun_find_featuremap_dims
+from typing import Dict, List, Tuple, Union
+
+import torch
+from omegaconf import DictConfig
+from torch import Tensor, nn
+
+from .timm import TimmFeatureExtractor, TimmFeatureExtractorParams
+from .torchfx import TorchFXFeatureExtractor, TorchFXFeatureExtractorParams
+
+
+class FeatureExtractor(nn.Module):
```

Comment:
i find it weird to have behavior/interfaces defined in `__init__.py` 
idk if there is a practical advantage/disavantage but i don't recall to have seen this in other projects